nixos/modules/services/monitoring/prometheus/exporters/mktxp.nix

```diff
@@ -0,0 +1,236 @@
+            certificate = false; # Certificates metrics
+
+            container = false; # Containers metrics
+
+            remote_dhcp_entry = null; # An MKTXP entry to provide for remote DHCP info / resolution
+            remote_capsman_entry = null; # An MKTXP entry to provide for remote capsman info
+
+            use_comments_over_names = true; # when available, forces using comments over the interfaces names
+            check_for_updates = false; # check for available ROS updates
+          };
+        };
+      in
+      pkgs.writeText "mktxp.conf" (
+        toPythonINI (lib.attrsets.recursiveUpdate defaultConfiguration cfg.configuration)
+      );
+in
+{
+  port = 49090;
+
+  extraOpts = {
+    package = mkPackageOption pkgs "mktxp" { };
+
+    configFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        The path to the mktxp configuration ini file. For more info see
+        https://github.com/akpw/mktxp
+
+        Note that setting this option is mutually exclusive with setting
+        the `configuration` option.
+      '';
+    };
+
+    configuration = mkOption {
+      type = types.attrsOf types.anything;
+      default = { };
+      description = ''
+        The mktxp configuration. To pass credentials, you can either
+
+        1. set the `username` and `password` options, or
+        2. use the `credentials_file` option to point to a YAML file
+          containing `username` and `password` keys.
+
+        For more info see https://github.com/akpw/mktxp
+
+        Note that setting this option is mutually exclusive with setting
+        the `configFile` option.
+      '';
+      example = {
+        Sample-Router = {
+          enabled = true;
+          hostname = "192.168.0.1";
+          port = 8728;
+
+          username = "admin";
+          password = "password";
+
+          connections = true;
+          lte = true;
+          user = false;
+        };
+      };
+    };
+
+    configWritable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to make the `/var/run/mktxp/mktxp/mktxp.conf` configuration
+        file writable. This can be useful if you need the `mktxp` executable
+        being able to write to the configuration file.
+
+        However, bear in mind that it will be overwritten at every start of the
+        service.
+      '';
+    };
+  };
+
+  serviceOpts = {
+    # The `mktxp` CLI requires `which` to properly initialize
+    # https://github.com/akpw/mktxp/blob/f2299ee5c264bb1a3d1d2bb781a4e63495e9c641/mktxp/cli/options.py#L233
+    path = [ pkgs.which ];
+
+    # NOTE: mktxp unfortunately does not support passing a custom
+    # configuration file or a custom config directory, it always reads what is
+    # present in $HOME/mktxp/mktxp.conf, so we populate a dynamics user's
+    # home directory with a symlink to the pure store configuration file.
```

Comment:
I think it might be possible to simplify this a lot if you were to bindmount (see [`systemd.exec(5)`](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#BindPaths=)) a derivation over the directory where it looks for config.